test/features/caseList.feature

```diff
@@ -8,3 +8,72 @@ Feature: Set of Scenarios testing the functionality of the Case List table
     Given a case type exists with case reference configured in the case list results
     And there are cases listed on the case list page for that case type
     Then the case reference is displayed in the case list results with hyphens
+
+  @pagination
+  Scenario: No Previous button on pagination when on page 1
```

Comment:
considering that ui func tests are  a bit slow, would it make sense to combine some of these scenarios into 1 or 2?
For example 1 scenario where you start on page 1 and verify all the stuff on page 1 then in the same scenario move to other pages and do the various verifications etc..

Reply:
Appreciate your concern about speed of tests however it's not good practice to bundle many things into a single test, if the first 'thing' you are testing fails you will have no knowledge  that the next few things in the test are broken or not as the test has failed early. 

These tests take 1min 38secs to run when I ran against AAT and this feature file would also be running in parallel with other features so not a huge hit on execution time. However, I understand the bigger picture that speed of tests is important and that we will get to a point where they will start to take too long but rather than changing the format of out tests there are other things we can do to help with execution time (eg run more in parallel, use cookie injection to bypass login form) which we can look into when needed

Reply:
rules for unit tests don't always apply to UI func tests. UI func tests usually are more coarse grained than unit tests. You can easily bundle all those into 1 test, if it fails you can understand where it fails from the step in error. 
Consider that in a previous project we used to write 1 scenario that was testing a multi page functionality and that means it had to navigate between various pages and verify multiple stuff. If we breaked that into single little tests it'll be much less understandable and valuable